module/x/gravity/types/interfaces.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	proto "github.com/gogo/protobuf/proto"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+)
+
+// EthereumSignature represents one validtors signature for a given
+// outgoing ethereum transaction
+type EthereumSignature interface {
+	proto.Message
+
+	GetSigner() common.Address
+	GetSignature() hexutil.Bytes
+	GetStoreIndex() []byte
+	Validate() error
+}
+
+// _ EthereumSignature = &ContractCallTxSignature{}
+// _ EthereumSignature = &BatchTxSignature{}
+// _ EthereumSignature = &SignerSetUpdateTxSignature{}
+
+// EthereumEvent represents a event from the gravity contract
+// on the counterparty ethereum chain
+type EthereumEvent interface {
+	proto.Message
+
+	GetNonce() uint64
+	GetEthereumHeight() uint64
+	Hash() tmbytes.HexBytes
+	Validate() error
+}
+
+// _ EthereumEvent = &DepositEvent{}
+// _ EthereumEvent = &WithdrawEvent{}
+// _ EthereumEvent = &ContractCallTxExecutedEvent{}
+// _ EthereumEvent = &CosmosERC20DeployedEvent{}
+
+type OutgoingTx interface {
+	// NOTE: currently the function signatures here don't match, figure out how to do this properly
+	// maybe add an interface arg here and typecheck in each implementation?
+	GetCheckpoint([]byte) ([]byte, error)
```

Comment:
In my evidence slashing PR, I simply replaced the errors with panics to make everything conform to `GetCheckpoint([]byte) ([]byte)`. There is no need to return an error because the problem can only be caused by programmer error.

Reply:
Awesome!

Reply:
Come to think of it, while we are renaming things, Checkpoint is a pretty bad name.

In the case of signer sets, we use a hash of the signer set in the Ethereum contract to save on gas instead of storing the whole thing. That's where the word "Checkpoint" came from. In the context of any other tx other than signer set updates, it is meaningless. What it really is, is a Digest, although that's kind of a weird word too.

Reply:
Could just call it a hash